interop/javapoet/src/test/kotlin/com/squareup/kotlinpoet/javapoet/PoetInteropTest.kt

```diff
@@ -139,23 +139,23 @@ class PoetInteropTest {
       JWildcardTypeName.supertypeOf(String::class.java),
     )
     assertThat(inKType.toJTypeName()).isEqualTo(superJType)
-    assertThat(superJType.toKTypeName()).isEqualTo(inKType)
+    assertThat(superJType.toKTypeName().toString()).isEqualTo(inKType.toString())
```

Comment:
This test was only passing because `TypeName` equality was previously equality of `toString()` representation. 

In actuality, the `inKType` created through `typeNameOf<GenericType<in String>>()` gets an `enclosedType` while the version obtained through `JParameterizedTypeName` does not.

Reply:
Maybe a way to highlight this to consumers would be:

`equals()` and `hashCode()` of `TypeName` are now more strict. The previous behavior can be retained by comparing `toString()` representations.

Reply:
SGTM!